src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -406,24 +408,58 @@ def _handleNoStatusUpdate(self, workload, request_args, dn):
+        #       which, by itself is a destructive function, which changes `request_args` with the following line:
+        #           request_name = request_args.pop("RequestName")
+        #       So if we call this function more than once inside the same request instance on every next call it will raise:
+        #           `KeyError: 'RequestName'`.
+        #       The first time it was called was during the validation of the REST call.
+        if 'RequestName' not in reqArgsDiff:
+            reqArgsDiff['RequestName'] = workload.name()
+        _, reqArgsDiff = validate_request_update_args(reqArgsDiff, self.config, self.reqmgr_db_service, None)
+
+        def _reduceReport(report):
+            if reduce(lambda x, y: x == y == 'OK' and 'OK', report, 'OK'):
+                return 'OK'
+            return False
+
+        if not reqArgsDiff:
+            cherrypy.log("Nothing to be changed at this stage")
+            return _reduceReport(report)
+
+        reqArgsNothandled = []
+        for reqArg in reqArgsDiff.copy():
+            if reqArg == 'RequestPriority':
+                validate_request_priority(reqArgsDiff)
+                # must update three places: GQ elements, workload_cache and workload spec
+                self.gq_service.updatePriority(workload.name(), reqArgsDiff['RequestPriority'])
+                report.append(self.reqmgr_db_service.updateRequestProperty(workload.name(), reqArgsDiff, dn))
+                workload.setPriority(reqArgsDiff['RequestPriority'])
+                workload.saveCouchUrl(workload.specUrl())
+                cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), reqArgsDiff['RequestPriority']))
+            elif workqueue_stat_validation(reqArgsDiff):
+                report.append(self.reqmgr_db_service.updateRequestStats(workload.name(), reqArgsDiff))
+                cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), reqArgsDiff))
+            elif reqArg == "SiteWhitelist":
+                report.append(self.reqmgr_db_service.updateRequestProperty(workload.name(), reqArgsDiff, dn))
+                workload.setSiteWhitelist(reqArgsDiff["SiteWhitelist"])
+                cherrypy.log('Updated SiteWhitelist of "{}", with:  {}'.format(workload.name(), reqArgsDiff['SiteWhitelist']))
+            elif reqArg == "SiteBlacklist":
+                report.append(self.reqmgr_db_service.updateRequestProperty(workload.name(), reqArgsDiff, dn))
+                workload.setSiteBlacklist(reqArgsDiff["SiteBlacklist"])
+                cherrypy.log('Updated SiteBlacklist of "{}", with:  {}'.format(workload.name(), reqArgsDiff['SiteBlacklist']))
+            else:
+                reqArgsNothandled.append(reqArg)
+                cherrypy.log("Uunhandled argument for no-status update: %s" % reqArg)
+
+        if reqArgsNothandled:
```

Comment:
I tend to disagree... This would be unneeded only if we decide NOT to act on finding a request parameter change which we do not support and silently accept the user's call and never giving a feedback on what we have dropped. Of course this is one way to go.

Reply:
If you raise this exception, it will get mapped to a http status code >=400, which is error. 
So we cannot execute some actions in the workload, and in the end report to the end user that his http request failed. It is inconsistent and misleading, at the very least.

Reply:
ok, give me your suggestion on this. How would you want to address the situation. To me it is all good. just let me know what is your preference.

Reply:
Yes, now I see it can have value in knowing which parameters from the user were dropped, so it's Okay to write a record of those in the logs.

However, this behavior is different than the behavior currently implemented in master. In current master, we only fail the HTTP call if the user didn't provide any of the expected parameters (RequestPriority or stats). While with this PR, it fails for any unsupported parameter provided by the end user (even worse, which could be after persisting changes for supported parameters).

Reply:
@amaltaro , this is addressed in my  next PR. See my comment above where I point to the lines of change in my next PR. We will again warn the user for the unsupported parameters requested, but we will do it as early as the workload update in memory and we will never persist anything in the database, because it will happen not here, but rather where it need be - in the code for workload update.